stock_move_accounting_date/models/stock_move.py

```diff
@@ -0,0 +1,21 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    accounting_date = fields.Date(
+        compute="_compute_accounting_date",
+        store=True,
+    )
+
+    @api.depends("accounting_date")
+    def _compute_accounting_date(self):
+        for line in self:
+            if not line.picking_id.accounting_date:
+                line.accounting_date = line.picking_id.date_done
+            else:
+                line.accounting_date = line.picking_id.accounting_date
```

Comment:
```suggestion
        for line in self:
            line.accounting_date = line.picking_id.date_done
            if line.picking_id.accounting_date:
                line.accounting_date = line.picking_id.accounting_date
```

Reply:
there's a "date" field on stock move which showing scheduled date of stock picking if the picking not done yet, and showing effective date (date_done).
as per discussed on meet room, im going to change it to the "date" field instead of date_done from picking.

Reply:
> there's a "date" field on stock move which showing scheduled date of stock picking if the picking not done yet, and showing effective date (date_done).

Beware that this may not the case with stock moves from MOs (please check the behavior), and we might have to handle this separately upon confirming the need with the client.

Reply:
@yostashiro san, yes i checked the behavior and confirm that the stock move of the product (not the component) from MOs has no done date so in the stock move menu the date remain the same with the date_planned_start.